### ISSUE TITLE
add method 'is_sunlit' to class Geocentric

### ISSUE THREAD

**Issue by lcx366** [skyfielders/python-skyfield#543]
1. At present, the `.is_sunlit` method or attribute can only be applied to a satellite object, is it possible to apply to any `Geocentric` instances ?
2. `Geocentric` object only supports au as its unit. Can it be expanded with a unit parameter, such as `unit='km'`?

**Comment by brandon-rhodes** [comment]
> At present, the .is_sunlit method or attribute can only be applied to a satellite object, is it possible to apply to any Geocentric instances ?

The intention was for the method to apply broadly. What error are you getting when you apply it to other geocentric instances?

> Geocentric object only supports au as its unit. Can it be expanded with a unit parameter, such as unit='km'?

Instead of adding a unit parameter, it might make more sense to allow users to supply their own `Distance` for the position, since those objects already include a unit parameter. I wonder whether it would be better to add that to `build_position`, or the position constructor, or else whether there should be an ICRS reference frame in `framelib` which could be used with the `from_time_and_frame_vectors()` constructor? I will look at the options and see which seems like the best fit.

**Comment by lcx366** [comment]
My initial code is

```
from skyfield.api import load
from skyfield.positionlib import Geocentric

ts = load.timescale()
t = ts.utc(2014, 1, 20, 0, 20)

eph = load('de430.bsp')

x,y,z = 3.141e-5, 2.718e-5, 5.820e-5 # unit in au
pos = Geocentric([x, y, z])

print(pos.at(t).is_sunlit(eph))
```

it gives

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-15-b133b484b7df> in <module>
     10 pos = Geocentric([x, y, z])
     11 
---> 12 print(pos.at(t).is_sunlit(eph))

AttributeError: 'Geocentric' object has no attribute 'at'
```

Later, I modified the code and get a result

```
from skyfield.api import load
from skyfield.positionlib import Geocentric

ts = load.timescale()
t = ts.utc(2014, 1, 20, 0, 20)

eph = load('de430.bsp')

x,y,z = 3.141e-5, 2.718e-5, 5.820e-5 # unit in au
pos = Geocentric([x, y, z],t=t)

print(pos.is_sunlit(eph))
```

True

When defining the position of a target in ICRF, the three cartesian components must be converted from kilometers to au. For long-distance stars, it is natural to use au, but for near-Earth objects such as satellites, km is more appropriate.

**Comment by brandon-rhodes** [comment]
> Later, I modified the code and get a result

I'm glad you figured out where to put the time `t` — yes, it should be provided as a constructor argument.

> When defining the position of a target in ICRF, the three cartesian components must be converted from kilometers to au. For long-distance stars, it is natural to use au, but for near-Earth objects such as satellites, km is more appropriate.

What if I pivoted the position class to allow a `Distance` to be provided for the x,y,z? Then you could do:
```
pos = Geocentric(Distance(km=[x,y,z]), t=t)
```
And meters could be provided as well. Let me know if that would make your use case easier!

**Comment by lcx366** [comment]
`pos = Geocentric(Distance(km=[x,y,z]), t=t)` doesn't work, and  `pos = Geocentric(Distance(km=np.array([x,y,z])).au, t=t)` works very well.  `pos = Geocentric([x, y, z],t=t,unit='km')` may  look more concise, if the parameter `position_au` in class `Geocentric` is replaced by `position` and `unit`.

**Comment by brandon-rhodes** [comment]
> `pos = Geocentric(Distance(km=[x,y,z]), t=t)` doesn't work…

I agree — but I was trying to suggest, if Skyfield were improved so that it *did* work, then I think that would also solve your problem, but without introducing a new `unit` parameter that would duplicate code that already exists in the `Distance` object?

> if the parameter position_au in class Geocentric is replaced by position and unit.

Alas, parameters can never be renamed without breaking older code that already uses their names. But I could, while leaving the name `position_au`, still auto-detect when a `Distance` is provided instead. (I could also make a separate constructor, and maybe should do more thinking to compare the two possibilities.)
